control-plane/agents/src/bin/core/controller/reconciler/node/nexus.rs

```diff
@@ -143,21 +159,66 @@ async fn check_and_drain_node(context: &PollContext, node_spec: &NodeSpec) -> Po
             }
         };
     }
-    // Change the node state to "drained"
     if !move_failures {
-        if let Err(error) = context
-            .specs()
-            .set_node_drained(context.registry(), node_spec.id())
-            .await
-        {
-            tracing::error!(
-                %error,
-                node.id = node_id.as_str(),
-                "Failed to set node to state drained"
-            );
-            return PollResult::Err(error);
+        // All volumes on the node are republished.
+        // Determine whether we can mark the node as drained.
+        // If not, the next reconciliation loop will check again.
+        let mut can_set_drained = true;
+
+        let drainingvolumes = context.specs().get_node_draining_volumes(&node_id).await?;
+        for (vi, starttime) in drainingvolumes {
+            let elapsed = starttime.elapsed();
+            let mut remove_vol_from_list = true;
+            if elapsed.is_ok()
+                && elapsed.unwrap() < Duration::from_secs(DRAINING_VOLUME_TIMOUT_SECONDS)
+                && context.specs().volume_rsc(&vi).is_some()
```

Comment:
You can probably use `if let Some(elapsed) = ` to reduce multiple condition checks.

Reply:
I tried that but rust complains: error[E0658]: `let` expressions in this position are unstable
   --> control-plane/agents/src/bin/core/controller/reconciler/node/nexus.rs:172
I'll see if I can find something that works.